packages/rust/server/src/services/password.rs

```diff
@@ -0,0 +1,20 @@
+use argon2::password_hash::SaltString;
+use argon2::{Argon2, PasswordHash, PasswordHasher, PasswordVerifier};
+
+use rand_core::OsRng;
+
+pub fn hash(password: &str) -> Result<String, argon2::password_hash::Error> {
+    let byte_password = password.as_bytes();
+    let salt = SaltString::generate(&mut OsRng);
+    let argon2 = Argon2::default();
+    Ok(argon2.hash_password(byte_password, &salt)?.to_string())
+}
+
+pub fn verify(hash: &str, password: &str) -> Result<bool, argon2::password_hash::Error> {
```

Comment:
Why use Result and bool, can we just use a Result<(),()> or just a bool?

Reply:
This function errors if the hash is in an improper format.  The caller might want to distinguish between the successful false case and the error case.